internal/v1/handler_compose_image.go

```diff
@@ -180,6 +190,90 @@ func buildRepositories(arch *distribution.Architecture, imageType ImageTypes) []
 	return repositories
 }
 
+func (h *Handlers) buildRepositorySnapshots(ctx echo.Context, arch *distribution.Architecture, imageType ImageTypes, snapshotDate *string) ([]composer.Repository, error) {
+	date, err := time.Parse(time.RFC3339, *snapshotDate)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Snapshot date %s is not in RFC3339 format", *snapshotDate))
+	}
+
+	repoURLs := []string{}
+	for _, r := range arch.Repositories {
+		contains := len(r.ImageTypeTags) == 0
+		for _, it := range r.ImageTypeTags {
+			if it == string(imageType) {
+				contains = true
+				break
+			}
+		}
+		if contains {
+			repoURLs = append(repoURLs, *r.Baseurl)
+		}
+	}
+
+	repoUUIDs := []string{}
```

Comment:
Maybe

```
var repoUUIDs []string{}
```

as append does initialize by default, but no biggie. Ideally I would call `make` to pre-allocate to the correct size but this is fine.